[DOCS] Improve best practice docs and add version matrix (#4297)
sabbelasichon/typo3-rector: docs/best_practice_guide.md

### TLDR;

- apply older or current version rulesets first (if you're going from v10 to v12, apply v7/v8/v9/v10 sets first)

> - if possible start with version 2 of typo3-rector though as it has the bugfixes and all rules from version 10 to 13 of TYPO3

Reviewer: ```suggestion
- if possible start with version 2 of typo3-rector because it has the bugfixes and all rules from version 10 to 13 of TYPO3
```

];
```

So to migrate your TCA in `Configuration/Override/` it is required to add those keys.

Also we can only migrate, when the type of the TCA column is known. For overrides this means you have to add it, e.g. when you extend the doktype items.

```php
$pages = [
    'columns' => [
        'doktype' => [
            'items' => [
                ['foo', 'foo']
            ],
        ],
    ],
];
```

> add `'type' => 'select` to migrate with rector

Reviewer: add new line before for better readability
